src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,270 @@
+package duke;
+import exceptions.DukeException;
+import exceptions.TaskNotExistException;
+import model.Deadline;
+import model.Event;
+import model.Task;
+import model.Todo;
+import service.Parser;
+import service.Storage;
+import service.TaskList;
+import ui.UI;
+
+import java.util.ArrayList;
+
+
+public class Duke{
+    
+    public enum Command {
+        Todo, Deadline, Event, Delete, Mark, Unmark, List, Bye, Unknown, Find;
+
+        public static Command fromString(String maybeCommand) {
+            try {
+                return Command.valueOf(maybeCommand);
+            } catch (Exception e) {
+                return Unknown;
+            }
+        }
+    }
+
+    /**
+     * Performs the appropriate action based on the user's input of the task.
+     * From adding of tasks which can be todos, deadlines and events, deleting and marking and unmarking them.
+     *
+     * @param userInput user's commands
+     * @param todos the list of tasks that are currently present.
+     * @param isVerbose helps to ensure whether we are preloading (includes the need to save the entries)
+     * @throws RuntimeException
+     */
+    public static void processCommand(String userInput, TaskList todos, Boolean isVerbose) throws RuntimeException {
```

Comment:
This is a massive method (line 39 - 218)! You could consider using a separate utility class to read and parse input, and separate methods to return the desired actions.